# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/cairo/cairo/src/cairo-svg-surface-private.h
# Commit: 3cdfb39376b7
# Full Hash: 3cdfb39376b736597fe658a322389117686b7e82
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-05-07 09:13:07
# Regressor Bug: 1892913
# File Overlap Count: 3
# Description:
#   Bug 1892913 - patch 1 - Update cairo source to 1.18.0 r=gfx-reviewers,lsalzman
#   
#   Wholesale replacement of files in gfx/cairo/cairo with contents of the 1.18.0 release tarball,
#   omitting subdirectories that are not part of the core library build.
#   
# ==============================================================================

diff -r 9daa351f875b -r 3cdfb39376b7 gfx/cairo/cairo/src/cairo-svg-surface-private.h
--- a/gfx/cairo/cairo/src/cairo-svg-surface-private.h	Mon May 06 19:22:35 2024 +0000
+++ b/gfx/cairo/cairo/src/cairo-svg-surface-private.h	Mon May 06 19:25:12 2024 +0000
@@ -44,40 +44,20 @@
 #include "cairo-svg.h"
 
 #include "cairo-surface-private.h"
-#include "cairo-surface-clipper-private.h"
 
-typedef struct cairo_svg_document cairo_svg_document_t;
-
-typedef struct _cairo_svg_source_surface {
-    cairo_hash_entry_t base;
-    unsigned int id;
-    unsigned char *unique_id;
-    unsigned long unique_id_length;
-} cairo_svg_source_surface_t;
-
-typedef struct cairo_svg_surface {
+struct _cairo_svg_surface_start {
     cairo_surface_t base;
 
-    cairo_content_t content;
-
-    double width;
-    double height;
-    cairo_bool_t surface_bounded;
-
-    cairo_svg_document_t *document;
+    cairo_bool_t force_fallbacks;
+};
 
-    cairo_output_stream_t *xml_node;
-    cairo_array_t	   page_set;
-    cairo_hash_table_t    *source_surfaces;
+static inline void
+_cairo_svg_surface_set_force_fallbacks (void *abstract_surface,
+					cairo_bool_t force_fallbacks)
+{
+    struct _cairo_svg_surface_start *surface = (struct _cairo_svg_surface_start *) abstract_surface;
 
-    cairo_surface_clipper_t clipper;
-    unsigned int clip_level;
-    unsigned int base_clip;
-    cairo_bool_t is_base_clip_emitted;
-
-    cairo_paginated_mode_t paginated_mode;
-
-    cairo_bool_t force_fallbacks;
-} cairo_svg_surface_t;
+    surface->force_fallbacks = force_fallbacks;
+}
 
 #endif /* CAIRO_SVG_SURFACE_PRIVATE_H */